### PR TITLE
#441 Add us beams example

### PR DESCRIPTION
**Pull request by gordon-n-stevenson**
This PR addresses adding the us_beams example to kwave-python. Just the ipython notebook for now and probably needs some small changes

### CODE REVIEW

**Comment by waltsims** [comment]
Hey Gordon,

Thanks for the PR!

**Comment by waltsims** [comment]
@gordon-n-stevenson  you can convert your notebook to a python file with [nbconvert](https://nbconvert.readthedocs.io/en/latest/usage.html#executable-script). Please do this and add a readme for your example before further review.

File: examples/us_defining_transducer/us_defining_transducer.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+# %%
+# define the input signal
+source_strength = 1e6
+tone_burst_freq = 0.5e6
+tone_burst_cycles = 5
+input_signal = tone_burst(1/kgrid.dt, tone_burst_freq, tone_burst_cycles)
+input_signal = (source_strength/ (medium.sound_speed * medium.density)) * input_signal
+
+
+# %%
```
[the remark this answers is not in the document]
Comment (waltsims) [reply examples/us_defining_transducer/us_defining_transducer.py:65]: You mean `# %%` correct?

File: examples/us_beam_patterns/README.md
[the remark this answers is not in the document]
Comment (waltsims) [reply examples/us_beam_patterns/README.md:1]: yes, a description of the example like the others have.

**Comment by waltsims** [comment]
closes #441

File: examples/us_beam_patterns/us_beam_patterns.ipynb
[the remark this answers is not in the document]
Comment (gordon-n-stevenson) [reply examples/us_beam_patterns/us_beam_patterns.ipynb:1]: I put a rather half baked version of this notebook up for review, apologies! The latest version should hopefully manage the USE_STATISTICS call a bit better. 

I had trouble using kwave's in build spect function and so did this quite hacky workaround using scipy's implementation and reshaping the data. 

Any more elegant solution to this issue appreciated!

**Comment by gordon-n-stevenson** [comment]
Latest iteration of this PR is ready for review - the READMEs and python files should be in the right format. One outstanding issue is the last few blocks of us_beam_patterns.py and checking @waltsims @djps  you are happy with my approach to get the amplitude spectrum out of the data. 

Ideally being able to use the in-built kwave [spect](https://k-wave-python.readthedocs.io/en/latest/_modules/kwave/utils/filters.html#spect) would be a better approach but I was struggling to get it working with the 3D data

**Comment by waltsims** [comment]
Hey Gordon, thanks for your changes. I'll have a look. If you had an issue with the kwave spect function, it would be helpful if you could open an Issue so we can try to address it! At a minimum improved documentation!

**Comment by gordon-n-stevenson** [comment]
Sure thing, I'll raise a MWE of the issue I had and try to explain - I
didn't spend too much time on it and was more looking to get things to
mirror the OG k-wave example.

**Comment by gordon-n-stevenson** [comment]
Just a nudge on this PR, checks are passed just needs review

**Comment by waltsims** [comment]
Hey Gordon thanks for the nudge. I was looking at your PR this morning and will finish soon. Do you mind me making direct changes?

**Comment by waltsims** [comment]
Hey Gordon,

I made a decent amount of progress reviewing your PR. I found some issues locally when running it, specifically when evaluating the output of `USE_STATISTICS=false` and running the Python file directly. I've gotten through most of the edits and can push them directly to your branch if that's ok with you.

Best,
Walter

**Comment by gordon-n-stevenson** [comment]
Great! Please do!

**Review by waltsims — APPROVED** [review]

**Comment by waltsims** [comment]
@gordon-n-stevenson please test my changes

**Comment by gordon-n-stevenson** [comment]
Hi @waltsims - only got around to testing now, my delightful children made me sick over the last few days. 

Running the us_beams_pattern notebook seems to be all working well!!

**Comment by waltsims** [comment]
Thanks for the PR @gordon-n-stevenson !

**Pull request merged**